src/Application/ConfigResolver.php

```diff
@@ -41,7 +41,7 @@ public static function resolve(array $config, string $directory): array
 
         foreach (self::$presets as $presetClass) {
             if ($presetClass::getName() === $preset && is_array($config)) {
-                $config = array_replace_recursive($presetClass::get(), $config);
+                $config = array_merge_recursive($config, $presetClass::get());
```

Comment:
https://github.com/nunomaduro/phpinsights/pull/111/files, @dol what do you think about this? I am not sure that the  custom config should override. What's the problem with the merge?

Reply:
I see where the problem comes from: array_merge_recursive replaces the same key. For numeric keys this is a problem => https://www.php.net/manual/en/function.array-merge-recursive.php#106985

The best approach to fix this issue might be to define first, what are the expectations of an user might be. IMHO a custom config should be able to add elements to the preset. This can be fixed with a better merge strategy/function. Just copy/paste a recursive array merge function from the php.net or from https://api.drupal.org/api/drupal/includes!bootstrap.inc/function/drupal_array_merge_deep/7.x . Don't add an additional dependency to fix this problem => 'left-pad' NPM apocalypse 
CC @olivernybroe: Sorry for introducing this issue.

Reply:
@dol no worries. Just glad if we can find a solution for this. 
If we are going to write our own merge strategy, then it might make sense to make it more unique depending on what key you are changing. So for example with the `exclude` array, let's say I use Laravel preset, but one I want to not exclude one of the excludes in the preset. Then I can write `"resources" => false`, which would mean exclusion of resources should not happen.

Reply:
@dol @olivernybroe Can you validate this pull request? https://github.com/nunomaduro/phpinsights/pull/147